driver/queries.c

```diff
@@ -1217,18 +1248,38 @@ SQLRETURN EsSQLBindCol(
 	RET_STATE(stmt->hdr.diag.state);
 }
 
+static SQLRETURN set_row_diag(esodbc_desc_st *ird,
+	esodbc_state_et state, const char *msg,
+	SQLULEN pos, SQLINTEGER colno)
+{
+	esodbc_stmt_st *stmt = HDRH(ird)->stmt;
+	SQLWCHAR wbuff[SQL_MAX_MESSAGE_LENGTH], *wmsg;
+	int res;
+
+	if (ird->array_status_ptr) {
+		ird->array_status_ptr[pos] = SQL_ROW_ERROR;
+	}
+	if (msg) {
+		res = ascii_c2w((SQLCHAR *)msg, wbuff, SQL_MAX_MESSAGE_LENGTH - 1);
+		wmsg = 0 < res ? wbuff : NULL;
```

Comment:
updated.
I was avoiding the potentially superfluous assignment (in case `msg != NULL` and `0 < res`), but that's a thing the compiler can easily optimize away and this is anyways executed on the error case, so simplification is welcome here.
thanks.